examples/sqlalchemy/app.py

```diff
@@ -5,59 +5,54 @@
 import orm
 from connexion import NoContent
 
-db_session = None
-
 
 def get_pets(limit, animal_type=None):
-    q = db_session.query(orm.Pet)
-    if animal_type:
-        q = q.filter(orm.Pet.animal_type == animal_type)
-    return [p.dump() for p in q][:limit]
+    with db_sess_mkr() as db_session:
+        q = db_session.query(orm.Pet)
+        if animal_type:
+            q = q.filter(orm.Pet.animal_type == animal_type)
+        return [p.dump() for p in q][:limit]
 
 
 def get_pet(pet_id):
-    pet = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
-    return pet.dump() if pet is not None else ("Not found", 404)
+    with db_sess_mkr() as db_session:
+        pet = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
+        return pet.dump() if pet is not None else ("Not found", 404)
 
 
 def put_pet(pet_id, pet):
-    p = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
-    pet["id"] = pet_id
-    if p is not None:
-        logging.info("Updating pet %s..", pet_id)
-        p.update(**pet)
-    else:
-        logging.info("Creating pet %s..", pet_id)
-        pet["created"] = datetime.datetime.utcnow()
-        db_session.add(orm.Pet(**pet))
-    db_session.commit()
-    return NoContent, (200 if p is not None else 201)
+    with db_sess_mkr() as db_session:
+        p = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
+        pet["id"] = pet_id
+        if p is not None:
+            logging.info("Updating pet %s..", pet_id)
+            p.update(**pet)
+        else:
+            logging.info("Creating pet %s..", pet_id)
+            pet["created"] = datetime.datetime.utcnow()
+            db_session.add(orm.Pet(**pet))
+        db_session.commit()
+        return NoContent, (200 if p is not None else 201)
 
 
 def delete_pet(pet_id):
-    pet = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
-    if pet is not None:
-        logging.info("Deleting pet %s..", pet_id)
-        db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).delete()
-        db_session.commit()
-        return NoContent, 204
-    else:
-        return NoContent, 404
+    with db_sess_mkr() as db_session:
+        pet = db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).one_or_none()
+        if pet is not None:
+            logging.info("Deleting pet %s..", pet_id)
+            db_session.query(orm.Pet).filter(orm.Pet.id == pet_id).delete()
```

Comment:
This should be just `db_session.delete(pet)`

Reply:
Right, I should have spotted that, will fix.

Reply:
Oh, and I noticed only after writing the review comment that the code was always like this. :-)